File: app/src/main/java/volovyk/guerrillamail/data/SingleEvent.java
package volovyk.guerrillamail.data;

public class SingleEvent<T> {
    private boolean hasBeenHandled = false;
    private final T content;

    public SingleEvent(T content) {
        this.content = content;
    }

    public boolean hasBeenHandled() {
        return hasBeenHandled;
    }

    public T getContentIfNotHandled() {
        if (hasBeenHandled) {
            return null;
        } else {
            hasBeenHandled = true;
            return content;
        }
    }
}

File: app/src/main/java/volovyk/guerrillamail/data/SingleEvent.kt
package volovyk.guerrillamail.data

class SingleEvent<T>(private val content: T) {
    private var hasBeenHandled = false
    fun hasBeenHandled(): Boolean {
        return hasBeenHandled
    }

    val contentIfNotHandled: T?
        get() = if (hasBeenHandled) {
            null
        } else {
            hasBeenHandled = true
            content
        }
}